r/tests/testthat/test-schema.R

```diff
@@ -0,0 +1,34 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+context("test-schema")
+
+test_that("reading schema from raw vector", {
+  batch <- record_batch(tibble::tibble(x = 1:10))
+  bytes <- write_record_batch(batch, raw())
```

Comment:
It isn't obvious to me that this would write a complete stream (maybe this should be called `write_record_batch_stream`). In Python we have a `RecordBatch.serialize` method that produces a single Message

https://github.com/apache/arrow/blob/master/python/pyarrow/tests/test_ipc.py#L607

Reply:
Yeah. At some point we need to sit down and have a chat about ipc api in this package.